### PR TITLE
feat: [PL-38908]: Unit tests for Fav utils classes and adding indexes.

### PR DESCRIPTION
**Pull request by boopesh007**
## Describe your changes
Added Unit tests for Utils classed and added index. 
Refactoring resource type from object to String

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48682)

### CODE REVIEW

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
trigger uts

File: 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  @Test
+  @Owner(developers = BOOPESH)
+  @Category(UnitTests.class)
+  public void testValidConnectorFavorite() throws IOException {
+    Call userCall = mock(Call.class);
+    when(userClient.getUserById(userId)).thenReturn(userCall);
+    when(userCall.execute()).thenReturn(Response.success(new RestResponse(of(UserInfo.builder().build()))));
+    Optional<ConnectorResponseDTO> connectorResponseDTO =
+        of(ConnectorResponseDTO.builder()
+                .connector(ConnectorInfoDTO.builder().connectorConfig(NexusConnectorDTO.builder().build()).build())
```
Comment (gn-harness) [inline 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesValidatorTest.java:87]: none of these tests are needed

Comment (gn-harness) [reply 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesValidatorTest.java:87]: You should write tests on the actual contract. These are tests on the implementation details

i.e. You try creating a favorite and that should return InvalidException or whatever, if invalid entity is passed in.

File: 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/entities/BUILD.bazel [hunk 2]
```diff
@@ -13,7 +13,7 @@ java_library(
     deps = [
         "//:lombok",
         "//120-ng-manager/modules/favorites/contracts/src/main/java/io/harness/favorites:module",
-        "//960-persistence/src/main/java/io/harness/ng:module",
+        "//960-persistence:module",
```
[the remark this answers is not in the document]
Comment (boopesh007) [reply 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/entities/BUILD.bazel:16]: For adding the index, I want to reference the Mongo module in the persistence package, but it does not have a .bazel file, hence the up hierarchy, I had to reference the whole package.

File: 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesValidatorTest.java (hunk 1 above)
Comment (boopesh007) [reply 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesValidatorTest.java:87]: The actual tests on the contract are also there @gn-harness, it was merged in the last PR. I had the same thought as yours but there was a comment to cover unit tests for these as well. cc: @nishantsaini20 @rijajoo

**Comment by boopesh007** [comment]
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

Create one more test with multiple favorites
</blockquote></details>

Done

File: 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesResourceUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    assertThat(favoriteResponse.getCreated()).isNotNull();
+    assertThat(favoriteResponse.getFavorite()).isEqualTo(favoriteDTO);
+  }
+
+  @Test
+  @Owner(developers = BOOPESH)
+  @Category(UnitTests.class)
+  public void testToFavoriteResponseList() {
+    List<FavoriteResponse> favoriteResponseList =
+        favoritesResourceUtils.toFavoriteResponse(Collections.singletonList(favorite));
```
[the remark this answers is not in the document]
Comment (boopesh007) [reply 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesResourceUtilsTest.java:93]: Done

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
trigger ut1

**Comment by boopesh007** [comment]
trigger uts

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]

File: 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/remote/AccountFavoritesApiImpl.java [hunk 4]
```diff
@@ -58,7 +56,7 @@ public Response deleteAccountFavorite(
   }
 
   @Override
-  public Response getAccountFavorites(String userId, String harnessAccount, FavoritesResourceType resourceType) {
```
Comment (nishantsaini20) [inline 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/remote/AccountFavoritesApiImpl.java:61]: Why these reverts related to `FavoritesResourceType`?

Comment (nishantsaini20) [inline 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/remote/AccountFavoritesApiImpl.java:61]: Why these reverts related to `FavoritesResourceType`?

File: 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/services/impl/FavoritesServiceImpl.java [hunk 5]
```diff
@@ -53,10 +52,9 @@ public Favorite createFavorite(FavoriteDTO favoriteDTO, String accountIdentifier
   }
 
   @Override
-  public List<Favorite> getFavorites(String accountIdentifier, String orgIdentifier, String projectIdentifier,
-      String userId, FavoritesResourceType resourceTypeDTO) {
-    ResourceType resourceType =
-        resourceTypeDTO != null ? EnumUtils.getEnum(ResourceType.class, resourceTypeDTO.toString()) : null;
+  public List<Favorite> getFavorites(
+      String accountIdentifier, String orgIdentifier, String projectIdentifier, String userId, String resourceTypeDTO) {
```
Comment (nishantsaini20) [inline 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/services/impl/FavoritesServiceImpl.java:56]: Why list call has `resourceTypeDTO`? If it is required then why `orgIdentifier` and `projectIdentifier` are required as separate arguments. They can be part of DTO itself.

File: 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/services/impl/FavoritesServiceImpl.java [hunk 6]
```diff
@@ -74,9 +72,8 @@ public List<Favorite> getFavorites(
 
   @Override
   public void deleteFavorite(String accountIdentifier, String orgIdentifier, String projectIdentifier, String userId,
-      FavoritesResourceType resourceTypeDTO, String resourceId) throws InvalidRequestException {
-    ResourceType resourceType =
-        resourceTypeDTO != null ? EnumUtils.getEnum(ResourceType.class, resourceTypeDTO.toString()) : null;
+      String resourceTypeDTO, String resourceId) throws InvalidRequestException {
```
Comment (nishantsaini20) [inline 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/services/impl/FavoritesServiceImpl.java:75]: Same here. Why is resourceTypeDTO required in delete call?

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

Shouldn't the order of fields be as, accountIdentifier->userIdentifier->org->proj->resourceType->id?
</blockquote></details>

Done.

**Comment by boopesh007** [comment]
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

Was this not changed to FavoritesResourceType in earlier PR? Why this reverted back?
</blockquote></details>

Its because UI code generation is failing when we reference the FavoritesResourceType in the queryParams.
cc: @avinashmadhwani02

**Comment by boopesh007** [comment]
*[`120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/remote/AccountFavoritesApiImpl.java` line 0 at r4](https://reviewable.io/reviews/harness/harness-core/48682#gh-1219008124:-NXK_dXq--T1pG_ER1tb:b-dn6go6) ([raw file](https://github.com/harness/harness-core/blob/e8c7094ee185f86dbc0d46e0a7e3255ea7d97b16/120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/remote/AccountFavoritesApiImpl.java#L0)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

Why these reverts related to `FavoritesResourceType`?
</blockquote></details>

Its because UI code generation is failing when we reference the FavoritesResourceType in the queryParams.
cc: @avinashmadhwani02

**Comment by boopesh007** [comment]
*[`120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/services/impl/FavoritesServiceImpl.java` line 56 at r4](https://reviewable.io/reviews/harness/harness-core/48682#gh-1219012066:-NXK_hbO1p26hm3_jaTB:b5t4gye) ([raw file](https://github.com/harness/harness-core/blob/e8c7094ee185f86dbc0d46e0a7e3255ea7d97b16/120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/services/impl/FavoritesServiceImpl.java#L56)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Why list call has `resourceTypeDTO`? If it is required then why `orgIdentifier` and `projectIdentifier` are required as separate arguments. They can be part of DTO itself.
</blockquote></details>

Renamed it

**Review by gn-harness — APPROVED** [review]

**Comment by boopesh007** [comment]
trigger uts

File: 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesValidatorTest.java (hunk 1 above)
Comment (nishantsaini20) [reply 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/utils/FavoritesValidatorTest.java:87]: These test not validating anything. When a specific validator doesn't return an entity then `validateFavoriteEntry` is expected to throw an exception and should what these tests should be validating.

**Review by nishantsaini20 — DISMISSED** [review]

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
trigger uts

File: 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/remote/FavoritesOrgApiImplTest.java [hunk 7]
```diff
@@ -126,11 +124,10 @@ public void testDeleteOrgScopedFavorite() {
   @Owner(developers = BOOPESH)
   @Category(UnitTests.class)
   public void testDeleteOrgScopedFavoriteInvalidResourceTypeThrowException() {
-    FavoritesResourceType favoritesResourceType = FavoritesResourceType.fromValue("Random");
     doThrow(new InvalidRequestException("Please provide a valid resource Type"))
         .when(favoriteService)
-        .deleteFavorite(accountId, orgId, null, userId, favoritesResourceType, resourceId);
-    Response response = orgFavoriteApi.deleteOrgFavorite(orgId, userId, accountId, favoritesResourceType, resourceId);
+        .deleteFavorite(accountId, orgId, null, userId, "Random", resourceId);
```
[the remark this answers is not in the document]
Comment (boopesh007) [reply 120-ng-manager/modules/favorites/src/test/java/io/harness/favorites/remote/FavoritesOrgApiImplTest.java:129]: Including the changes for this in the subsequent PR @nishantsaini20

**Comment by boopesh007** [comment]
trigger uts

File: 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/utils/FavoritesValidator.java [hunk 8]
```diff
@@ -113,8 +113,10 @@ public void validateFavoriteEntry(FavoriteDTO favoriteDTO, String accountIdentif
     }
     if (!resourceExist) {
       log.error("The Resource with ID {} and type {}, which is being marked as favorite does not exist in account {}",
-          favoriteDTO.getResourceId(), favoriteDTO.getResourceType(), accountIdentifier);
-      throw new InvalidRequestException("The resource does not exist");
+              favoriteDTO.getResourceId(), favoriteDTO.getResourceType(), accountIdentifier);
+      throw new InvalidRequestException(String.format(
+              "The Resource with ID %s and type %s, which is being marked as favorite does not exist in account %s",
```
Comment (nishantsaini20) [inline 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/utils/FavoritesValidator.java:118]: Wrap value of ID and type in square brackets 
`"The Resource with ID [%s] and type [%s], which is being marked as favorite does not exist"`

**Review by nishantsaini20 — DISMISSED** [review]

File: 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/utils/FavoritesValidator.java (hunk 8 above)
Comment (boopesh007) [reply 120-ng-manager/modules/favorites/src/main/java/io/harness/favorites/utils/FavoritesValidator.java:118]: Will incorporate in the upcoming PR.

**Comment by boopesh007** [comment]
trigger uts

**Review by nishantsaini20 — APPROVED** [review]

**Review by gn-harness — APPROVED** [review]

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
trigger uts

**Comment by boopesh007** [comment]
trigger ut5

**Pull request merged**